proto/iceflow.proto

```diff
@@ -43,3 +43,28 @@ message RepartitionResponse {
   uint32 lower_partition_bound = 2;
   uint32 upper_partition_bound = 3;
 }
+
+service NodeExecutor {
+  /// Sends a report about a congestion on an edge within the Iceflow system
+  /// to the NodeExecutor.
+  rpc ReportCongestion(CongestionReportRequest) returns (CongestionReportResponse);
+}
+
+message CongestionReportRequest {
+  /// The specifier of the congested edge.
+  string edge_name = 1;
+
+  /// An identifier of the cause for the congestion.
+  CongestionReason congestion_reason = 2;
+  // TODO: Should the amount of congestion be quantified somehow?
+}
+
+message CongestionReportResponse {
+  // TODO: Does the NodeExecutor need to report something to the Node Instance?
+}
+
+enum CongestionReason {
+  // TODO: Discuss which kind of reasons we want to define here
```

Comment:
Those two should be sufficient for now, considering that it is the producer that sends these congestion notifications, and the producer doesn't really have more information than "my queue is filling up" (processing) or "i received an NDN congestion notification" (network).

Reply:
Thank you for the feedback on this :) I removed the TODO and added some additional documentation to individual reasons.